### PR TITLE
fix(ME-1894): use outer tls connection for connectorAuth and inner tls connection for e2e encryption

### PR DESCRIPTION
**Pull request by btoonk**
# Description

use outer tls connection for connectorAuth and inner tls connection for e2e encryption.

Fixes # (ME-1894)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Tested locally

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Review by atoonk — APPROVED** [review]

File: cmd/client/tls/tls.go [hunk 1]
```diff
@@ -79,7 +79,8 @@ var clientTlsCmd = &cobra.Command{
 				}
 
 				go func() {
-					conn, err := tls.Dial("tcp", fmt.Sprintf("%s:%d", hostname, info.Port), &tlsConfig)
+					var conn net.Conn
```
Comment (adrianosela) [inline cmd/client/tls/tls.go:82]: this change is redundant

File: internal/client/client.go [hunk 2]
```diff
@@ -883,9 +883,11 @@ func ConnectorAuthConnectWithConnV2(conn net.Conn, tlsConfig *tls.Config, connec
 			conn.Close()
 			return nil, fmt.Errorf("failed to write to proxy: %w", err)
 		}
-	}
 
-	return connectorConn, nil
+		return conn, nil
```
Comment (adrianosela) [inline internal/client/client.go:887]: inverted logic reads better.

```
connectorConn := tls.Client(conn, tlsConfig)
if err := connectorConn.Handshake(); err != nil {
	return nil, fmt.Errorf("failed to authenticate to connector: %w", err)
}

if !connectorAuthOnly {
    return connectorConn 
}

_, err := conn.Write([]byte("BORDER0-CLIENT-CONNECTOR-AUTHENTICATED"))
if err != nil {
	conn.Close()
	return nil, fmt.Errorf("failed to write to proxy: %w", err)
}

return conn
```

**Review by adrianosela — CHANGES_REQUESTED** [review]
comments left, but looks good

File: internal/client/client.go (hunk 2 above)
Comment (adrianosela) [reply internal/client/client.go:887]: `connectorAuthOnly` variable name is confusing too

**Review by adrianosela — APPROVED** [review]

File: internal/client/client.go (hunk 2 above)
Comment (btoonk) [reply internal/client/client.go]: agree, changed the logic

**Pull request merged**